pkg/prowgen/generators.go

```diff
@@ -51,6 +51,17 @@ func MakeTest(ctx *ProwContext) *Job {
 					Memory: "4Gi",
 				},
 			},
+			Lifecycle: &Lifecycle{
```

Comment:
We are mixing pointers and values here, for consistency's sake we should probably only use values

Reply:
If a non pointer is used then the value can never be unset/nil - some of our jobs don't use a lifecycle hook, and I think those should omit this field together in those cases